lib/saml_idp/xml_security.rb

```diff
@@ -236,8 +255,9 @@ def verify_signature(base64_cert, sig_alg, signature, canon_string, soft)
         cert = OpenSSL::X509::Certificate.new(cert_text)
         signature_algorithm = algorithm(sig_alg)
 
-        if signature_algorithm != SamlIdp.config.algorithm
-          return false if soft
+        if !soft && signature_algorithm != SamlIdp.config.algorithm
+          # remove comment when we are sure this wont break partners
+          # return false if soft
```

Comment:
I feel like we couldn't use this line exactly as-is, because the `!soft` means that `soft` will *never* be true here?

I feel like in the philosophy of "comments are lies waiting to happen" I would just remove this? The goal looks like it's a reminder of where to pick up for the next round of things?

Reply:
i waffled about including it for the reasons you mentioned, but i think it's important because if we forget to include it when we actually turn the error on in a few weeks, since without it, we will start returning this error to partners, and we don't want to do that yet

Reply:
to be clearer, when we know it's safe to turn on, the plan is to  remove the if `if !soft` part of the conditional

Reply:
ah! maybe I'd move the comment outside the if-block like `remove if soft when we know it's ready` or something

Reply:
okay great! updated in https://github.com/18F/saml_idp/pull/126/commits/6ff881f2dc9643344a75430bdfd6f516becc5f33